qiskit_ibm_runtime/session.py

```diff
@@ -301,15 +302,36 @@ def from_id(
             session_id: the id of the session to be created. This must be an already
                 existing session id.
             service: instance of the ``QiskitRuntimeService`` class.
-            backend: instance of :class:`qiskit_ibm_runtime.IBMBackend` class or
-                string name of backend.
+            backend (DEPRECATED): instance of :class:`qiskit_ibm_runtime.IBMBackend` class or
+                string name of backend. Either ``service`` or ``backend`` must be given.
+
+         Raises:
+            IBMInputValueError: If given `session_id` does not exist. or the backend passed in does
+                not match the original session backend.
 
         Returns:
             A new Session with the given ``session_id``
 
         """
         if backend:
             deprecate_arguments("backend", "0.15.0", "Sessions do not support multiple backends.")
+            if isinstance(backend, IBMBackend):
+                backend = backend.name
+
+        if service:
+            response = service._api_client.session_details(session_id)
+            if response:
+                session_backend = response.get("backend_name")
+                if backend and backend != session_backend:
+                    raise IBMInputValueError(
+                        f"The session_id {session_id} was created with backend {session_backend}, "
+                        f"but backend {backend} was given."
+                    )
+                backend = session_backend
+            else:
```

Comment:
Does the server just return an empty response if session is not found? I'd think it'd raise an 404 error.

Reply:
When `session.details` was first implemented, we decided to use a `try/except` because of the differences between the US and DE environments 

https://github.com/kt474/qiskit-ibm-runtime/blob/8723ccf827fd7b43bf653d76e9c254ad00f9be4b/qiskit_ibm_runtime/api/rest/runtime_session.py#L57-L63

So `None` will be returned if the session is not found

Reply:
DE is dead so does that change any minds on how to work this now?

Reply:
Yeah that logic should be updated, we can also remove the check [here](https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/qiskit_ibm_runtime/qiskit_runtime_service.py#L220)